Modules/Sources/Yosemite/PointOfSale/Eligibility/POSSystemStatusService.swift

```diff
@@ -53,6 +63,42 @@ public final class POSSystemStatusService: POSSystemStatusServiceProtocol {
     }
 }
 
+private extension POSSystemStatusService {
+    /// Updates or inserts system plugins in storage.
+    func upsertSystemPlugins(siteID: Int64, systemStatus: POSPluginEligibilitySystemStatus, in storage: StorageType) {
+        // Active and inactive plugins share identical structure, but are stored in separate parts of the remote response
+        // (and without an active attribute in the response). So we apply the correct value for active (or not)
+        let readonlySystemPlugins: [SystemPlugin] = {
+            let activePlugins = systemStatus.activePlugins.map {
+                $0.copy(active: true)
+            }
+
+            let inactivePlugins = systemStatus.inactivePlugins.map {
+                $0.copy(active: false)
+            }
+
+            return activePlugins + inactivePlugins
+        }()
+
+        let storedPlugins = storage.loadSystemPlugins(siteID: siteID, matching: readonlySystemPlugins.map { $0.name })
+        readonlySystemPlugins.forEach { readonlySystemPlugin in
+            // Loads or creates new StorageSystemPlugin matching the readonly one.
+            let storageSystemPlugin: StorageSystemPlugin = {
+                if let systemPlugin = storedPlugins.first(where: { $0.name == readonlySystemPlugin.name }) {
+                    return systemPlugin
+                }
+                return storage.insertNewObject(ofType: StorageSystemPlugin.self)
+            }()
+
+            storageSystemPlugin.update(with: readonlySystemPlugin)
+        }
+
+        // Removes stale system plugins.
+        let currentSystemPlugins = readonlySystemPlugins.map(\.name)
+        storage.deleteStaleSystemPlugins(siteID: siteID, currentSystemPlugins: currentSystemPlugins)
```

Comment:
🗒️ This is mostly the same as the upsert logic in `SystemStatusStore` https://github.com/woocommerce/woocommerce-ios/blob/0ea5d32119755276bd283d57eb5b52b0bbdd5f9e/Modules/Sources/Yosemite/Stores/SystemStatusStore.swift#L96-L129

In WOOMOB-859, I plan to refactor the upsert logic for reuse with some enhancements to ensure we're matching plugins by the plugin path instead of name.